.github/workflows/update-data-r.yml

```diff
@@ -0,0 +1,106 @@
+# Update /data when files in /data-raw are changed
+# This workflow runs R code to regenerate data in /data when files in /data-raw are changed.
+# Once the update is complete, the changes are committed.
+# If 'commit-directly' is set to false (the default), a pull request will be opened 
+# to the branch where the workflow was triggered. The PR will contain the commit with updated data.
+# If 'commit-directly' is set to true, changes are pushed directly to the triggering branch.
+# Note that the job will fail if force pushing is required.
+on:
+ workflow_call:
+   secrets:
+     PAT:
+       required: false
+   inputs:
+     commit-directly:
+       required: false
+       type: boolean
+       default: false
+
+# Give the fewest permissions possible. content and pull-requests are necessary.
+# https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token
+permissions:
+  contents: write
+  pull-requests: write
+
+# Cancel runs happening simultaneously on the same branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+name: update data
+
+jobs:
+  update-data-r:
+    name: update R data from /data-raw
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.PAT }}
+    steps:
+      - name: setup env using GITHUB_TOKEN, if no pat
+        if: env.GITHUB_PAT == ''
+        run: echo "GITHUB_PAT=${{ secrets.GITHUB_TOKEN}}" >> "$GITHUB_ENV"
+
+      - name: checkout, make changes and submit as pr on new branch
+        if: inputs.commit-directly == false
+        uses: actions/checkout@v4
+        
+      - name: checkout, make changes directly
+        if: inputs.commit-directly == true
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+      
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: |
+            devtools
+
+      - name: Run R code to update files in /data
+        run: |
+          devtools::install()
+          r_scripts <- list.files("data-raw", pattern = "\\.R$", full.names = TRUE)
+          if (length(r_scripts) > 0) {
+            for (script in r_scripts) {
+              system(paste("Rscript ", shQuote(script)))
```

Comment:
I am confused 😕 why you would use a system call to run Rscript instead of just sourcing the function within R here?